detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedPrivateMemberSpec.kt

```diff
@@ -1054,6 +1054,20 @@ class UnusedPrivateMemberSpec : Spek({
             assertThat(subject.compileAndLintWithContext(env, code)).isEmpty()
         }
 
+        it("does not report used plus operator in methods with +=") {
+            val code = """
+                import java.util.Date
+                class Foo {
+                    var number: Float? = null
+                    fun foo() {
+                        number += 3f
+                    }
+                    private operator fun Float?.plus(other: Float?) = if (this == null && other == null) null else this ?: 0f + (other ?: 0f)
```

Comment:
Detekt strives to keep the test code as simple as possible in order to make it more readable and maintainable.
The test snippets should be as simple as possible and test only the rule under test.

```suggestion
                class Test {
                    fun f() {
                        var number: Int = 0
                        number += 1
                        number -= 1
                        number *= 1
                        number /= 1
                        number %= 1
                    }
                    private operator fun Int.plus(other: Int) = 1
                    private operator fun Int.minus(other: Int) = 2
                    private operator fun Int.times(other: Int) = 3
                    private operator fun Int.div(other: Int) = 4
                    private operator fun Int.rem(other: Int) = 5
                    private operator fun Int.mod(other: Int) = 6
                }
```

Reply:
b974091 that's a good point. I had to keep the nullable type, as kotlin would otherwise give precedence to the operator function defined in `Int`. I also didn't add `mod` as it would be effectively unused (`rem` would be used instead). I preferred `rem` over `mod` as the latter is deprecated.